lib/AST/NameLookup.cpp

```diff
@@ -1295,8 +1272,15 @@ bool AbstractStorageDecl::isSetterAccessibleFrom(const DeclContext *DC) const {
   // accessibility, it is not set.
   if (hasStorage() && !isSettable(nullptr))
     return true;
-  
-  return checkAccessibility(DC, getDeclContext(), getSetterAccessibility());
+
+  auto setterAccess = getFormalSetterAccess(DC);
+  if (!DC)
+    return setterAccess >= Accessibility::Public;
+
+  auto scope = getDeclContext()->getAccessScope(DC, setterAccess,
+    /* useNominalTypeAccessibility */ false);
```

Comment:
You're giving me more credit than I deserve; I was asking about both cases.

For what it's worth, test/SILGen/witness_accessibility.swift is wrong—you can't (today) satisfy a public requirement with something that depends on a private protocol, and most times the compiler will reject it ([SR-697](https://bugs.swift.org/browse/SR-697)). But fixing that will probably break code, so okay.